armi/reactor/reactors.py

```diff
@@ -2280,17 +2286,21 @@ def processLoading(self, cs, dbLoad: bool = False):
                 for a in self.getAssemblies(includeAll=True):
                     a.makeAxialSnapList(self.refAssem)
 
-            if not cs["inputHeightsConsideredHot"]:
-                runLog.header(
-                    "=========== Axially expanding all (except control) assemblies from Tinput to Thot ==========="
-                )
-                axialExpChngr = AxialExpansionChanger(cs["detailedAxialExpansion"])
-                for a in self.getAssemblies():
-                    if not a.hasFlags(Flags.CONTROL):
-                        axialExpChngr.setAssembly(a)
-                        axialExpChngr.expansionData.computeThermalExpansionFactors()
-                        axialExpChngr.axiallyExpandAssembly(thermal=True)
-                axialExpChngr.manageCoreMesh(self.parent)
+            for a in self.getAssemblies(includeAll=True):
+                assemsToAxiallyExpand.append(a)
+
+        if not cs["inputHeightsConsideredHot"]:
+            runLog.header(
+                f"=========== Axially expanding {aExpStr} assemblies (except control) from Tinput to Thot ==========="
+            )
+            axialExpChngr = AxialExpansionChanger(cs["detailedAxialExpansion"])
+            for a in assemsToAxiallyExpand:
+                if not a.hasFlags(Flags.CONTROL):
+                    axialExpChngr.setAssembly(a)
+                    axialExpChngr.expansionData.computeThermalExpansionFactors()
+                    axialExpChngr.axiallyExpandAssembly(thermal=True)
+            axialExpChngr.manageCoreMesh(self.parent)
+            self.updateBlockBOLHeights()
```

Comment:
I think this should only be called on the core when you are not loading from DB. Loads from DB should just use what is stored in DB.

Reply:
Do we not want to expand the blueprints assemblies when loading from DB?

Reply:
you could move the update inside the for loop and it would work

Reply:
Addressed https://github.com/terrapower/armi/pull/823/commits/15499a11f58af5a0803e4645e818c488da9b0f4b